Source/Dafny/Compiler-java.cs

```diff
@@ -2087,6 +2087,16 @@ protected override void EmitActualTypeArgs(List<Type> typeArgs, Bpl.IToken tok,
             opString = "+";
           } else if (resultType is UserDefinedType && !resultType.IsIntegerType && !unsignedTypes.Contains(TypeName(e0.Type, null, tok)) && !unsignedTypes.Contains(TypeName(e1.Type, null, tok))) {
             opString = "+";
+            switch (TypeName(resultType, null, tok)) {
```

Comment:
Could you add a reference to somewhere explaining why we're casting here.  E.g. https://docs.oracle.com/javase/specs/jls/se7/html/jls-5.html#jls-5.6.2